nexus/inventory/src/collector.rs

```diff
@@ -368,6 +371,40 @@ impl<'a> Collector<'a> {
             ),
         }
     }
+
+    /// Collect inventory from about keepers from all `ClickhouseAdminKeeper`
+    /// clients
+    async fn collect_all_keepers(&mut self) {
+        let clients = self.keeper_admin_clients.take().unwrap();
+        for client in &clients {
+            self.collect_one_keeper(&client).await;
+        }
+    }
+
+    /// Collect inventory about one keeper from one `ClickhouseAdminKeeper`
+    async fn collect_one_keeper(
+        &mut self,
```

Comment:
This doesn't need to be a method on self I think, it could be a static method or just a top-level function.

Reply:
Agreed. I used @jgallagher's solution below here as well. Thank you again for the helpful input @sunshowers!